docs/secrets/Secrets.md

```diff
@@ -1,98 +1,217 @@
+
+- [Pull an Image from a Private Registry][k8s_docs-pull_image_private_registry]
+- [Add ImagePullSecrets to a service account][k8s_docs_add_imagepullsecrets_to_service_account]
 
 
 ### Jenkinsfile Runner Image
 
-> **Note:** Does not apply for GCP. If the registry and the K8s cluster are hosted in the same Google account (region?) then no pull secret is required.
+If the image of the Jenkinsfile Runner container resides in a private registry, an image pull secret must be configured by a Steward operator.
+By default the Jenkinsfile Runner image is hosted in a public repository on Docker Hub, which does not require authentication when pulling.
+
+__TODO__: How to configure the image pull secret for the Jenkinsfile Runner image
+
+
+### Pipeline Custom Pod Images
+
+Pipelines may start additional pods to execute steps in containers, either via the Jenkins Kubernetes plugin or by other means using the Kubernetes API.
+If the required container images are hosted in private registries, image pull secrets must be provided by the respective Steward client (on behalf on end users).
+
+A Steward PipelineRun resource object specifies the names of all image pull secrets required for that pipeline run:
+
+```yaml
+apiVersion: steward.sap.com/v1alpha1
+kind: Tenant
+spec:
+    ...
+    imagePullSecrets:
+    - secret1
+    - secret2
+    - secret3
+```
+
+If no image pull secrets are required, `spec.imagePullSecrets` can be omitted or have an empty list value.
+
+The given secrets are standard Kubernetes `v1/Secret` resource objects that must exist in the same namespace as the PipelineRun object that references them.
+The secrets must be of the correct type (`kubernetes.io/dockerconfigjson`) to be usable for Kubernetes as image pull secret.
+Besides that there are no further requirements like special annotations or labels.
+
+When a pipeline gets executed in a transient sandbox namespace, the secrets listed in `spec.imagePullSecrets` of the corresponding PipelineRun resource object are copied to the sandbox namespace with a different name.
+The Kubernetes service account of the Jenkinsfile Runner container has all those secrets attached as default image pull secrets.
+Therefore, they will be used automatically when that service account creates pods based on images from private registries.
+The pipeline and/or tools used by the pipeline to start additional pods are not required to specify image pull secrets in each pod specification, although this is still possible.
+
+__:warning: Warning:__ Any code that gets executed by a pipeline AND has access to the Kubernetes service account token can read all image pull secrets! This is especially important to consider if untrusted code may get executed, e.g. a pipeline processing pull requests from untrusted users.
+
+- Any code running in the Jenkinsfile Runner container, because this container has the service account token mounted.
```

Comment:
It's not clear that the bullet points belong to the warning. Add an introducing sentence.

Reply:
I added the intro sentence. However, it would be nice to have the complete warning inside a panel. But I didn't find a way (using HTML) where markdown still works inside.